website/pages/blog/type-safe-your-api.mdx

```diff
@@ -0,0 +1,132 @@
+
+## Introduction
+
+Developers who have experience working with REST API often encounter challenges such as lack of type safety, manual response parsing, and the absence of a unified interface for endpoints. These issues can hinder the efficiency and reliability of REST API development.
+
+In this blog post, we will explore a powerful tool called feTS that addresses these problems and provides solutions for more effective REST API development.
+
+## What Is feTS?
+feTS, which stands for "Fetch" + "TypeScript" is a robust tool designed specifically for TypeScript projects. It simplifies the development and consumption of REST API while ensuring type safety. By leveraging OpenAPI specifications and TypeScript decorators, feTS allows developers to build and interact with REST API more efficiently, reducing the likelihood of runtime errors and improving code quality. Importantly, feTS is fully compatible with any existing REST API.
+
+## How I Used Web API before feTS?
+
+In this blog post, we will use the Spotify API as an example, but you can apply these principles to any REST API of your choice. You can find the complete code example [here](https://github.com/TuvalSimha/feTS-example).
+
+Spotify provides a developer-friendly API that is easy to use. Here's a example of how to retrieve your top 5 tracks from the last 30 days using the Spotify API:
+
+1. Create a Spotify account and retrieve your access token from [here](https://developer.spotify.com/documentation/web-api/reference/get-users-top-artists-and-tracks?type=artists&time_range=short_term&limit=5&offset=0).
+2. Create a fetchWebApi function that fetches data from the API and returns it as JSON:
+    
+```typescript
+const token = 'YOUR_TOKEN_HERE';
+
+async function fetchWebApi(endpoint: string, method: string, body: undefined) {
+  const res = await fetch(`https://api.spotify.com/${endpoint}`, {
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+    method,
+    body: JSON.stringify(body),
+  });
+  
+  return await res.json();
+}    
+```
+
+3. Call the fetchWebApi function with the desired endpoint:
+```typescript
+async function getTopTracks() {
+  // Endpoint reference: https://developer.spotify.com/documentation/web-api/reference/get-users-top-artists-and-tracks
+  return (await fetchWebApi(
+    'v1/me/top/tracks?time_range=short_term&limit=5',
+    'GET',
+    undefined
+  )).items;
+}
+```
+4. Lastly, you can use the data from the API:
+```typescript
+async function main() {
+  const topTracks = await getTopTracks();
+  console.log(topTracks);
+}
+
+main();
+```
+
+While this approach works, it has some drawbacks:
+
+- The fetchWebApi function lacks type safety, making it uncertain what data will return from the API.
+- You need to manually specify the parameters of the fetchWebApi function.
+- The fetchWebApi function allows calling any endpoint, even if it's not supported by the API.
+- The topTracks variable lacks type safety according to the API documentation.
```

Comment:
```suggestion
- The `topTracks` variable lacks type safety according to the API documentation.
```

Maybe something simpler like: "We don't know the type of `topTracks`" or that it's `any`